### PR TITLE
I and II

### PR DESCRIPTION
**Pull request by invegat**

### CODE REVIEW

File: algorithms/egg.js [hunk 1]
```diff
@@ -0,0 +1,58 @@
+function getRandomInt(max) {
```
Comment (ely-alamillo) [inline algorithms/egg.js]: Awesome, you are the only one that did this program separate.

File: algorithms/exercises.md [hunk 2]
```diff
@@ -0,0 +1,84 @@
+# Exercise I
+**a**. O(N^3^)
```
Comment (ely-alamillo) [inline algorithms/exercises.md]: this one is `O(n)`

File: algorithms/exercises.md [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+**b**. O(log(N))  Any time the power of N is less than 1 (here 0.5) it's called logrithmic.
+**c**. O(N)  N^1.5^ rounded down to N
```
Comment (ely-alamillo) [inline algorithms/exercises.md]: this one is `O(sqrt(n))`

File: algorithms/exercises.md [hunk 4]
```diff
@@ -0,0 +1,84 @@
+# Exercise I
+**a**. O(N^3^)
+**b**. O(log(N))  Any time the power of N is less than 1 (here 0.5) it's called logrithmic.
+**c**. O(N)  N^1.5^ rounded down to N 
+**d**. O(N)  N^1.5^ rounded down to N
```
Comment (ely-alamillo) [inline algorithms/exercises.md]: this one `O(n*log(n))`

File: theory/regex_test.js [hunk 5]
```diff
@@ -0,0 +1,38 @@
+//  `antelope` and `antelopes` 
+console.log(
+`
+antelope rocks out
+antelopes rock out
+`.match( /(antelope(?:s)?)/g));
```
Comment (ely-alamillo) [inline theory/regex_test.js]: nice, could have just been `(antelopes? rocks? out)`

File: algorithms/exercises.md (hunk 4 above)
Comment (invegat) [reply algorithms/exercises.md]: ok thanks

File: algorithms/exercises.md (hunk 3 above)
Comment (invegat) [reply algorithms/exercises.md]: ok thanks

**Pull request closed without merging**